packages/next/next-server/lib/router/router.ts

```diff
@@ -676,16 +678,18 @@ export default class Router implements BaseRouter {
     })
   }
 
-  _getStaticData = (asPath: string, _cachedData?: object): Promise<object> => {
-    let pathname = parse(asPath).pathname
+  _getStaticData = (asPath: string, cache?: boolean): Promise<object> => {
+    let { pathname, query } = parse(asPath, true)
     pathname = toRoute(!pathname || pathname === '/' ? '/index' : pathname)
 
-    return process.env.NODE_ENV === 'production' &&
-      (_cachedData = this.sdc[pathname])
-      ? Promise.resolve(_cachedData)
+    return process.env.NODE_ENV === 'production' && this.sdc[pathname]
+      ? Promise.resolve(this.sdc[pathname])
       : fetch(
-          // @ts-ignore __NEXT_DATA__
-          `/_next/data/${__NEXT_DATA__.buildId}${pathname}.json`
+          formatWithValidation({
+            // @ts-ignore __NEXT_DATA__
+            pathname: `/_next/data/${__NEXT_DATA__.buildId}${pathname}.json`,
+            query,
```

Comment:
Why is this passed unconditionally? It'll cause a cache miss at CDN level for SSG pages.

Reply:
Updated